### PR TITLE
cli: add check of successful upload for `upload-bin`

### PR DESCRIPTION
**Pull request by AliceInHunterland**
Refs #3658

### CODE REVIEW

File: cli/util/uploader.go [hunk 1]
```diff
@@ -581,11 +582,19 @@ func uploadObj(ctx context.Context, p *pool.Pool, signer user.Signer, owner util
...
 	_, err = writer.Write(objData)
 	if err != nil {
 		return fmt.Errorf("failed to write object data: %w", err)
 	}
+	err = writer.Close()
+	if err != nil {
+		return fmt.Errorf("failed to close object writer: %w", err)
+	}
+	res := writer.GetResult()
+	if res.StoredObjectID().Equals(oid.ID(emptyOid)) {
```
Comment (AnnaShaleva) [inline cli/util/uploader.go:602]: Why not just `oid.ID{}` instead of `oid.ID(emptyOid)`?

File: cli/util/convert.go [hunk 2]
```diff
@@ -90,6 +90,11 @@ func NewCommands() []*cli.Command {
 			Name:  "skip-blocks-uploading",
 			Usage: "Skip blocks uploading and upload only index files",
 		},
+		&cli.UintFlag{
+			Name:  "max-retries",
+			Usage: "Maximum number of retries of request",
```
Comment (AnnaShaleva) [inline cli/util/convert.go:95]: Maximum number of NeoFS request retries

File: cli/util/uploader.go [hunk 3]
```diff
@@ -50,6 +48,11 @@ const (
 	maxBackoff = 20 * time.Second
 )
 
+var (
+	// Maximum number of retries.
+	maxRetries int
+)
```
Comment (AnnaShaleva) [inline cli/util/uploader.go:54]: Don't make it global, it's a command parameter now.

File: cli/util/convert.go [hunk 4]
```diff
@@ -90,6 +90,11 @@ func NewCommands() []*cli.Command {
 			Name:  "skip-blocks-uploading",
 			Usage: "Skip blocks uploading and upload only index files",
 		},
+		&cli.UintFlag{
+			Name:  "max-retries",
```
Comment (AnnaShaleva) [inline cli/util/convert.go:94]: Just `retries`.

File: cli/util/uploader.go (hunk 1 above)
Comment (AliceInHunterland) [reply cli/util/uploader.go:602]: Can be `oid.ID{}`, but `if res.StoredObjectID().Equals(oid.ID{})` check could be redundant.

File: cli/util/convert.go (hunk 2 above)
Comment (AliceInHunterland) [reply cli/util/convert.go:95]: we retry neo rpc too

File: cli/util/uploader.go (hunk 1 above)
Comment (AnnaShaleva) [reply cli/util/uploader.go:602]: > check could be redundant.

Why? Is it guaranteed by NeoFS SDK that this OID is always non-empty?

File: cli/util/convert.go (hunk 2 above)
Comment (AnnaShaleva) [reply cli/util/convert.go:95]: OK, then just `Maximum number of Neo/NeoFS node request retries`.

File: cli/util/uploader.go (hunk 1 above)
Comment (AliceInHunterland) [reply cli/util/uploader.go]: Not sure. But we even can calculate and set OID on our side before putting it.

Comment (AnnaShaleva) [reply cli/util/uploader.go]: > Not sure. 

Can you check it?

File: cli/util/uploader.go [hunk 5]
```diff
@@ -581,11 +582,19 @@ func uploadObj(ctx context.Context, p *pool.Pool, signer user.Signer, owner util
 	if err != nil {
 		return fmt.Errorf("failed to initiate object upload: %w", err)
 	}
-	defer writer.Close()
+
 	_, err = writer.Write(objData)
 	if err != nil {
 		return fmt.Errorf("failed to write object data: %w", err)
 	}
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply cli/util/uploader.go]:
```
 if err != nil {
                _ = writer.Close()
		return fmt.Errorf("failed to write object data: %w", err)
	}
```

File: cli/util/convert.go [hunk 6]
```diff
@@ -90,6 +90,17 @@ func NewCommands() []*cli.Command {
 			Name:  "skip-blocks-uploading",
 			Usage: "Skip blocks uploading and upload only index files",
 		},
+		&cli.UintFlag{
+			Name:  "retries",
+			Usage: "Maximum number of Neo/NeoFS node request retries",
+			Value: 5,
+			Action: func(context *cli.Context, u uint) error {
+				if u < 1 {
+					return cli.Exit("retries should be greater than 0", 1)
```
Comment (AnnaShaleva) [inline cli/util/convert.go:99]: Good check.

File: docs/neofs-blockstorage.md [hunk 7]
```diff
@@ -99,6 +99,7 @@ OPTIONS:
 --workers value                                                                Number of workers to fetch, upload and search blocks concurrently (default: 50)
 --searchers value                                                              Number of concurrent searches for blocks (default: 20)
 --skip-blocks-uploading                                                        Skip blocks uploading and upload only index files (default: false)
+--retries value                                                            Maximum number of Neo/NeoFS node request retries (default: 5)
```
Comment (AnnaShaleva) [inline docs/neofs-blockstorage.md:102]: Something has happened with the indent.

**Review by AnnaShaleva — COMMENTED** [review]
Otherwise ready for the merge.

**Comment by AnnaShaleva** [comment]
Rebase, please.

**Pull request merged**